src/main/java/net/io/kino/controller/OrderController.java

```diff
@@ -1,35 +1,47 @@
 package net.io.kino.controller;
 
+import net.io.kino.controller.converter.StringToDateConverter;
 import net.io.kino.controller.dto.OrderRequest;
-import net.io.kino.model.Order;
-import net.io.kino.model.Showtime;
+import net.io.kino.model.*;
+import net.io.kino.service.EmailSender;
 import net.io.kino.service.ReservationService;
+import net.io.kino.service.ShowtimeService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.server.ResponseStatusException;
 
+import java.sql.Time;
 import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
 @RestController
+@RequestMapping("/orders")
 public class OrderController {
 
     ReservationService reservationService;
+    ShowtimeService showtimeService;
 
     @PostMapping
     public Order addOrder(@RequestBody OrderRequest orderRequest) {
         return reservationService.createOrder(orderRequest.getTickets()
-                .stream().map(t -> t.convertToTicket(new Showtime())) //TODO: Get showtime from service
+                .stream().map(t -> t.convertToTicket(showtimeService.findShowtimeById(t.getShowtimeId())))
                 .collect(Collectors.toList()), orderRequest.getClient());
     }
 
     @PutMapping("/{id}")
     public ResponseEntity<HttpStatus> updateOrder(@PathVariable Long id, @RequestBody OrderRequest orderRequest) {
-        return new ResponseEntity<>(HttpStatus.OK);
+        Optional<Order> order = reservationService.findOrderById(id);
+        if (!order.isPresent()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        reservationService.confirmOrder(order.get());
```

Comment:
Z tego co tu widzę, to w tym momencie ten endpoint służy do potwierdzenia zamówienia, ale z definicji RESTa, PUT powinien aktualizować cały zasób. Też nie wiem w jakim momencie miałby być używany.

Reply:
Ale oni w sumie nie mieli kontrolerów pisać to chyba wywalone

Reply:
To usuńcie kod kontrollerowy, żeby Jackowi nie mieszać.

Reply:
cofnęliśmy wszystkie zmiany wykonane przez nas